src/table/utils/styling-utils.js

```diff
@@ -25,58 +27,78 @@ export const SELECTION_STYLING = {
   },
 };
 
-export function getColor(color = {}, defaultColor, theme) {
+export function getColor(defaultColor, theme, color = {}) {
   const resolvedColor = theme.getColorPickerColor(color);
   return !resolvedColor || resolvedColor === 'none' ? defaultColor : resolvedColor;
 }
 
 export const getAutoFontColor = (backgroundColor) =>
   isDarkColor(backgroundColor) ? STYLING_DEFAULTS.WHITE : STYLING_DEFAULTS.FONT_COLOR;
 
-export const getBaseStyling = (styleObj, theme) => ({
-  color: getColor(styleObj.fontColor, STYLING_DEFAULTS.FONT_COLOR, theme),
-  fontSize: styleObj.fontSize || STYLING_DEFAULTS.FONT_SIZE,
-  padding: styleObj.fontSize ? `${styleObj.fontSize / 2}px ${styleObj.fontSize}px` : STYLING_DEFAULTS.PADDING,
-});
+export const getBaseStyling = (styleObj, objetName, theme) => {
+  const backgroundColor = theme.getStyle('object', 'straightTable', 'backgroundColor');
+  const fontFamily = theme.getStyle('object', `straightTable.${objetName}`, 'fontFamily');
+  const color = theme.getStyle('object', `straightTable.${objetName}`, 'color');
+  const fontSize = theme.getStyle('object', `straightTable.${objetName}`, 'fontSize');
+
+  const baseStyle = {
+    backgroundColor,
+    fontFamily,
+    color: styleObj?.fontColor ? getColor(STYLING_DEFAULTS.FONT_COLOR, theme, styleObj?.fontColor) : color,
```

Comment:
can styleObj be undefined?

Reply:
it can. layout.components?.[0]?.header can be undefined